### PR TITLE
test: Small collection of miscellaneous test coverage fix-ups

### PR DESCRIPTION
**Pull request by lukelooker**
This was the by-product of auditing a random things on other PRs and picking up the left-over stashes into a small collection of coverage improvements.

### CODE REVIEW

File: packages/components-test-utils/src/snapshot.tsx [hunk 1]
```diff
@@ -26,14 +26,8 @@
...
 
 export const assertSnapshot = <T extends {}>(element: ReactElement<T>) => {
   expect(createWithTheme(element).toJSON()).toMatchSnapshot()
 }
-
-export const assertSnapshotShallow = <T extends {}>(
-  element: ReactElement<T>
-) => {
-  expect(shallowWithTheme(element)).toMatchSnapshot()
-}
```
Comment (lukelooker) [inline packages/components-test-utils/src/snapshot.tsx:39]: This is used _nowhere_

File: packages/components/src/Dialog/Dialog.test.tsx [hunk 2]
```diff
@@ -41,12 +41,32 @@ import {
...
     renderWithTheme(<Dialog content={<SimpleContent />} />)
     expect(screen.queryByText('Dialog content')).not.toBeInTheDocument()
   })
 
+  test('Placement functions', () => {
+    renderWithTheme(
+      <Dialog isOpen placement="top" content={<SimpleContent />} />
+    )
+    expect(screen.queryByText('Dialog content')).toBeInTheDocument()
+  })
```
Comment (lukelooker) [inline packages/components/src/Dialog/Dialog.test.tsx:55]: We never tested placement in Jest

File: packages/components/src/Form/Fields/Field.test.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { screen } from '@testing-library/react'
+import { Field } from './Field'
+
+describe('Field', () => {
+  test('Hidden labels', () => {
+    renderWithTheme(
+      <Field id="test" hideLabel label="hello!">
+        <input id="test" type="text" />
+      </Field>
+    )
```
Comment (lukelooker) [inline packages/components/src/Form/Fields/Field.test.tsx:38]: Hidden labels had no test coverage

File: packages/components/src/Form/Inputs/InputText/InputText.test.tsx [hunk 4]
```diff
@@ -116,6 +121,15 @@ describe('InputText', () => {
...
     })
 
+    test('Simple string', () => {
+      const { getByText } = renderWithTheme(
+        <InputText before="before" after="after" />
+      )
+
+      expect(getByText('before')).toBeVisible()
+      expect(getByText('after')).toBeVisible()
+    })
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/InputText/InputText.test.tsx:131]: There's a specific nuance in before/after where if it's a string it's presented with a wrapping `span` to apply type styles.

File: packages/components/src/Form/Inputs/InputTime/utils.ts [hunk 5]
```diff
@@ -30,18 +30,6 @@ import map from 'lodash/map'
...
-  if (format === '12h') {
-    if (hour === 0) {
-      return 12 // 12:00 am
-    } else if (hour > 12) {
-      return hour - 12 // 1:00 pm - 11:00 pm
-    }
-  }
-  return hour
-}
-
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/InputTime/utils.ts:44]: This is never used in this location

File: packages/components/src/Form/Inputs/InputTime/InputTime.tsx [hunk 6]
```diff
@@ -55,13 +55,13 @@ import {
...
 import { getAutoFocusProps } from '../InputProps'
+import { ValidationType } from '../../ValidationMessage'
 import {
   formatTimeString,
   TimeFormats,
   parseBase10Int,
   isValidTime,
-} from '../utils'
-import { ValidationType } from '../../ValidationMessage'
+} from './utils'
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/InputTime/InputTime.tsx:64]: Move utils into `InputTime` (they were sitting in `Inputs` before)

**Review by dbchristopher — APPROVED** [review]
✅

**Pull request merged**